src/codegen/interpreter/bytecode_builder.cpp

```diff
@@ -0,0 +1,1830 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// bytecode_builder.cpp
+//
+// Identification: src/codegen/interpreter/bytecode_builder.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/interpreter/bytecode_builder.h"
+
+#include <llvm/IR/InstIterator.h>
+#include <fstream>
+
+#include "codegen/codegen.h"
+#include "common/exception.h"
+
+namespace peloton {
+namespace codegen {
+namespace interpreter {
+
+BytecodeBuilder::BytecodeBuilder(const CodeContext &code_context,
+                                 const llvm::Function *function)
+    : bytecode_function_(function->getName().str()),
+      number_value_slots_(0),
+      number_temporary_value_slots_(0),
+      rpo_traversal_(function),
+      code_context_(code_context),
+      llvm_function_(function) {}
+
+BytecodeFunction BytecodeBuilder::CreateBytecodeFunction(
+    const CodeContext &code_context, const llvm::Function *function,
+    bool use_naive_register_allocator) {
+  BytecodeBuilder builder(code_context, function);
+  builder.AnalyseFunction();
+
+  if (use_naive_register_allocator)
```

Comment:
For these `if-else` statements, our style has been to use braces around each condition. I really don't care, but since the rest of the code uses it, better to be consistent. 

There is also mixed usage of braces/no braces in this file. Again, just be consistent.

Reply:
Sorry, I think I miss understood. Do you mean braces around the body of if-statements or parentheses around if conditions if they get chained with and/or?